controllers/update_pods.go

```diff
@@ -182,6 +199,10 @@ func getPodsToUpdate(logger logr.Logger, reconciler *FoundationDBClusterReconcil
 		updates[zone] = append(updates[zone], pod)
 	}
 
+	if maxUnavailablePods > 0 && unavailablePods >= maxUnavailablePods {
+		return nil, fmt.Errorf("unavailable Pods reached cluster.Spec.MaxUnavailablePods limit: %d", maxUnavailablePods)
+	}
+
```

Comment:
I believe the current logic is not doing what it proposes: `MaxUnavailablePods defines the maximum number or percent of pods that can be unavailable during the update process.`

But we are actually only checking if `unavailablePods >= maxUnavailablePods` before doing the updates. So we could hit a case where `maxUnavailablePods` is `5` and `unavailablePods` is `4`, this would mean `unavailablePods >= maxUnavailablePods` is false and we have an update budget of 1 Pod, but in the current implementation we are not verifying if the `updates` will only update `1` Pod and not more (otherwise we have more unavailable Pods than defined in `maxUnavailablePods`)